pkg/debug/checkups/gnome_extensions_linux.go

```diff
@@ -27,10 +30,6 @@ const (
 )
 
 func (c *gnomeExtensions) Name() string {
-	if runtime.GOOS != "linux" {
```

Comment:
do we need to get rid of this pattern because it won't compile without `GnomeExtensions` for other distros? Is it maybe cleaner to make the alternate versions of offending methods in pkg/allowedcmd and just have them return nil/error since they wouldn't ever run anyway

Reply:
I went back and forth on this a bit in earlier iterations of this PR, but landed here for compile safety -- if a command isn't available on a particular OS, then we don't want to be able to use it.

There were actually relatively few places I had to update to make this work -- I think that putting OS-specific commands in `_<os>.go` is actually closer to the pattern we use frequently, and `pkg/debug/checkups` was a departure from that pattern. There were just a few one-offs in the tables I had to update, plus here, plus one in the osquery log processing (which was my fault anyway 😅 ).

Reply:
ahh okay i see what you mean. this makes more sense to me then, thanks!

Reply:
Compiler enforced restrictions is the best. When I wrote the checkups, I did the lazy thing and added the ability to shortcut without needing the build tags. And given their use, it seemed okay. But now that we have allowed commands, we need to push that through here